contracts/ColonyFunding.sol

```diff
@@ -174,14 +172,23 @@ contract ColonyFunding is ColonyStorage, DSMath {
     emit RewardPayoutCycleStarted(globalRewardPayoutCount);
   }
 
-  function claimRewardPayout(uint256 _payoutId, uint256[7] _squareRoots, uint256 _userReputation, uint256 _totalReputation) public {
-    RewardPayoutCycle memory payout = rewardPayoutCycles[_payoutId];
+  function claimNextRewardPayout(uint256[7] _squareRoots, uint256 _userReputation, uint256 _totalReputation) public {
+    uint256 payoutId = add(userRewardPayoutCount[msg.sender], 1);
```

Comment:
We discussed this a few times already but just to reiterate. Inferring the `payoutId` this way has a problem for any user who "misses" a reward payout for any reason. Examples:
- `payoutId: 9` has finished and a user earns their first Colony tokens and deposit/lock them. Then `payoutId: 10` starts and user calls `claimNextRewardPayout` which infers the `payoutId` as `1` not `10`
- Token holder X successfully claims the first 10 reward payouts which takes `userRewardPayoutCount[X] = 10`. Then they unlock their tokens and sell them all making them ineligible for reward payouts. Another 3 reward payouts go by and  user X earns some colony tokens again which they deposit/lock. The current payout Id is `14` but the `userRewardPayoutCount` derived here `10+1=11`. The claim fails.

Reply:
You also have to add tests covering the above scenarios.

Reply:
Ok, so you mean, if the user was not eligible to claim first 9 payouts, **we** should increment the counter for him (in claimNextRewardPayout?), or you are simply stating that we should add tests for these scenarios?

Reply:
You cannot _just_ add tests for this scenario as this won't work with the current implementation. I am pointing out a problem that should be fixed and tested for.